sklearn/tests/test_discriminant_analysis.py

```diff
@@ -223,6 +225,59 @@ def test_lda_scaling():
                      'using covariance: %s' % solver)
 
 
+def test_lda_store_covariance():
+    # Test for slover 'lsqr' and 'eigen'
+    for solver in ('lsqr', 'eigen'):
+        clf = LinearDiscriminantAnalysis(solver=solver)
+        assert_true(hasattr(clf.fit(X, y), 'covariance_'))
```

Comment:
Please use `assert_warns` on the `hasattr` to check that this behaviour is deprecated appropriately.

Either do that here and remove the test below, or make this test only about non-deprecated behaviour and avoid this assertion with a comment saying that the opposite should be asserted once the deprecation is complete.